### PR TITLE
fix(state-viewer): print hash and memory usage of tree nodes

### PR DESCRIPTION
**Pull request by nikurt**
* makes `neard view-chain --format full` print hash and node's memory usage.
* adds a `print` sub-sub-sub-command to print the sub-tree contained in a state part: `./neard view-state state-parts --root-dir /tmp --shard-id 0 apply --part-id 42 --action print current`
* fixes `neard view-state-part dump` to print state record in a human readable way.

### CODE REVIEW

**Comment by nikurt** [comment]
cc: @Longarithm

**Comment by akhi3030** [comment]
assigning @Longarithm as the reviewer.

File: tools/state-viewer/src/state_parts.rs [hunk 1]
```diff
@@ -20,13 +21,21 @@ use std::path::{Path, PathBuf};
...
+    #[default]
+    Apply,
+    Validate,
+    Print,
+}
+
 #[derive(clap::Subcommand, Debug, Clone)]
 pub(crate) enum StatePartsSubCommand {
     /// Apply all or a single state part of a shard.
     Apply {
```
Comment (Longarithm) [inline tools/state-viewer/src/state_parts.rs:35]: Looks good, the only concern I have here is readability. When I read "apply", I assume that something is applied/changed, but now we can have validate and print actions which doesn't modify anything.

I suggest to rename parent action to `Load`, similarly to python `json` load & dump API, and have `LoadAction` and `load_state_parts` to show that we read state parts from somewhere. Inside we have `Apply` action which means that we also apply part to state, etc.

**Review by Longarithm — CHANGES_REQUESTED** [review]

File: tools/state-viewer/src/state_parts.rs [hunk 2]
```diff
@@ -270,38 +279,50 @@ fn apply_state_parts(
         assert!(part_id < num_parts, "part_id: {}, num_parts: {}", part_id, num_parts);
```
Comment (Longarithm) [inline tools/state-viewer/src/state_parts.rs:279]: in the comment above: "Applying state as seen at the beginning of the specified epoch." -> "Loading state as seen at the beginning of the specified epoch."

File: tools/state-viewer/src/state_parts.rs [hunk 3]
```diff
@@ -270,38 +279,50 @@ fn apply_state_parts(
...
+                    &part
+                ));
+                tracing::info!(target: "state-parts", part_id, part_length = part.len(), elapsed_sec = timer.elapsed().as_secs_f64(), "Validated a state part");
+            }
+            LoadAction::Print => {
+                print_state_part(&state_root, PartId::new(part_id, num_parts), &part)
+            }
         }
     }
     tracing::info!(target: "state-parts", total_elapsed_sec = timer.elapsed().as_secs_f64(), "Applied all requested state parts");
```
Comment (Longarithm) [inline tools/state-viewer/src/state_parts.rs:317]:
```suggestion
    tracing::info!(target: "state-parts", total_elapsed_sec = timer.elapsed().as_secs_f64(), "Loaded all requested state parts");
```

File: tools/state-viewer/src/state_parts.rs [hunk 4]
```diff
@@ -20,13 +21,21 @@ use std::path::{Path, PathBuf};
...
+}
+
 #[derive(clap::Subcommand, Debug, Clone)]
 pub(crate) enum StatePartsSubCommand {
-    /// Apply all or a single state part of a shard.
-    Apply {
-        /// If true, validate the state part but don't write it to the DB.
-        #[clap(long)]
-        dry_run: bool,
+    /// Load all or a single state part of a shard and perform an actions wth those parts.
```
Comment (Longarithm) [inline tools/state-viewer/src/state_parts.rs:34]:
```suggestion
    /// Load all or a single state part of a shard and perform an action wth those parts.
```

**Review by Longarithm — APPROVED** [review]

**Pull request merged**